File: April 22, 2026.js
function getCleanupScore(items) {

  const itemsList = {
    bottle: { baseValue: 10 },
    can: { baseValue: 6 },
    bag: { baseValue: 8 },
    tire: { baseValue: 35 },
    straw: { baseValue: 4 },
    cardboard: { baseValue: 3 },
    newspaper: { baseValue: 3 },
    shoe: { baseValue: 12 },
    electronics: { baseValue: 25 },
    battery: { baseValue: 18 },
    mattress: { baseValue: 38 }
  };

  let total = 0;
  let streak = 0;
  let itemCount = 0;
  let lastItem = "";

  for (let item of items) {

    itemCount += 1;
    let points = 0;

    if (Array.isArray(item) && item[0] == "rare") {
      points = item[1];

      if (itemCount % 5 == 0) {
        points *= (itemCount / 5) + 1;
      }

      streak = 0;
      lastItem = "";
      total += points;
      continue;
    }

    let itemValue = itemsList[item].baseValue;
    points = itemValue;

    if (item == lastItem) {
      streak += 1;
    } else {
      streak = 0;
    }

    points = itemValue + streak;
    if (itemCount % 5 == 0) {
      points *= (itemCount / 5) + 1;
    }

    lastItem = item;
    total += points;
  }
  
  return total;
}

File: April 22, 2026.py
def get_cleanup_score(items):
    items_list = {
        "bottle": {"base_value": 10},
        "can": {"base_value": 6},
        "bag": {"base_value": 8},
        "tire": {"base_value": 35},
        "straw": {"base_value": 4},
        "cardboard": {"base_value": 3},
        "newspaper": {"base_value": 3},
        "shoe": {"base_value": 12},
        "electronics": {"base_value": 25},
        "battery": {"base_value": 18},
        "mattress": {"base_value": 38}
    }

    total = 0
    streak = 0
    item_count = 0
    last_item = ""

    for item in items:
        item_count += 1
        points = 0

        if isinstance(item, list) and item[0] == "rare":
            points = item[1]

            if item_count % 5 == 0:
                points *= (item_count // 5) + 1

            streak = 0
            last_item = ""
            total += points
            continue

        item_value = items_list[item]["base_value"]
        points = item_value

        if item == last_item:
            streak += 1
        else:
            streak = 0

        points = item_value + streak

        if item_count % 5 == 0:
            points *= (item_count // 5) + 1

        last_item = item
        total += points

    return total